napari/_qt/qt_color_dialog.py

```diff
@@ -0,0 +1,257 @@
+        """
+        try:
+            _color = transform_color(color)[0]
+        except ValueError:
+            return self.color_changed.emit(self._color)
+        emit = np.any(self._color != _color)
+        self._color = _color
+        if emit or np.all(_color == TRANSPARENT):
+            self.color_changed.emit(_color)
+
+
+class QColorLineEdit(QLineEdit):
+    """A LineEdit that takes hex, rgb, or autocompletes common color names."""
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self._compl = QCompleter(list(get_color_dict()) + ['transparent'])
+        self._compl.setCompletionMode(QCompleter.InlineCompletion)
+        self.setCompleter(self._compl)
+        self.setTextMargins(2, 2, 2, 2)
+
+    def setText(self, color: ColorType):
+        """Set the text of the lineEdit using any ColorType.
+
+        Colors will be converted to standard SVG spec names if possible,
+        or shown as #RGBA hex if not.
+
+        Parameters
+        ----------
+        color : ColorType
+            Can be any ColorType recognized by our
+            utils.colormaps.standardize_color.transform_color function.
+        """
+        _rgb = transform_color(color)[0]
+        _hex = rgb_to_hex(_rgb)[0]
+        super().setText(hex_to_name.get(_hex, _hex))
+
+
+class CustomColorDialog(QColorDialog):
+    def __init__(self, parent=None):
+        super().__init__(parent=parent)
+        self.setObjectName('CustomColorDialog')
+
+    def keyPressEvent(self, event):
+        event.ignore()
+
+
+class QColorPopup(QtPopup):
+    """A QColorDialog inside of our QtPopup.
+
+    Allows all of the show methods of QtPopup (like show relative to mouse).
+    Passes through signals from the ColorDialogm, and handles some keypress
+    events.
+    """
+
+    currentColorChanged = Signal(QColor)
+    colorSelected = Signal(QColor)
+
+    def __init__(self, parent=None, initial=None):
+        super().__init__(parent)
+        self.setObjectName('QtColorPopup')
+        self.color_dialog = CustomColorDialog(self)
+
+        # native dialog doesn't get added to the QtPopup frame
+        # so more would need to be done to use it
+        self.color_dialog.setOptions(
+            QColorDialog.DontUseNativeDialog | QColorDialog.ShowAlphaChannel
+        )
+        layout = QVBoxLayout()
+        self.frame.setLayout(layout)
+        layout.addWidget(self.color_dialog)
+
+        self.color_dialog.currentColorChanged.connect(
+            self.currentColorChanged.emit
+        )
+        self.color_dialog.colorSelected.connect(self._on_color_selected)
+        self.color_dialog.rejected.connect(self._on_rejected)
+        self.color_dialog.setCurrentColor(QColor(initial))
+
+    def _on_color_selected(self, color):
+        self.colorSelected.emit(color)
+        self.close()
+
+    def _on_rejected(self):
+        self.close()
+
+    def keyPressEvent(self, event):
+        """Accept current color on enter, cancel on escape."""
```

Comment:
Very nice. =)

Can you add a Parameters section here, specifying the class of `event`? ie if I want to modify the behaviour of this function, how do I see what I can and can't do with `event`?